### ISSUE TITLE
`Credo.Check.Warning.MissedMetadataKeyInLoggerConfig` should by default allow Erlang's metadata keys

### ISSUE THREAD

**Issue by hauleth** [rrrene/credo#1046]
### Environment

* Credo version (`mix credo -v`):
* Erlang/Elixir version (`elixir -v`):
* Operating system:

### What were you trying to do?

Insert code like:

```elixir
Logger.debug(%{foo: 1}, report_cb: &__MODULE__.func/1)
```

### Expected outcome

Credo reports nothing

### Actual outcome

Credo complains that `:report_cb` key is not used in production because it is not present in the logger configuration. Though it *is* used as it is part of the message formatting process that is happening in the Erlang and Elixir default formatters.

Ideally `Credo.Check.Warning.MissedMetadataKeyInLoggerConfig` should be opt-in, as logger metadata can also be used for filtering, and that is impossible to deduce by static analysis.

**Comment by rrrene** [comment]
What would be a list of "Erlang's metadata"? Would [these keys](https://www.erlang.org/doc/man/logger_formatter#type-config) be safe to ignore?

**Comment by hauleth** [comment]
The thing is, that there can be custom backend that does anything with the metadata, or there can be `report_cb` function that will utilise that metadata as well. So there is no way to programatically check whether metadata is used or not.

**Comment by rrrene** [comment]
Sure, but "there _could_ be specific circumstances which make this check unsuitable for individual users" is a good argument to make all the checks opt-in, isn't it?

I added `:report_cb` to the ignored metadata keys in the check. Let me know if you think we should add more (since the title of this issue was about that) 👍

**Comment by rrrene** [comment]
FYI: This is part of `1.7.2-rc.3` :+1:

**Comment by rrrene** [comment]
This is part of Credo `1.7.2`. :+1:
